### DISCUSSION TITLE
Getting Error on Pnp MOdern Search WebPart when Org Member but Not Admin

### DISCUSSION THREAD

**Discussion by RamBalani** [microsoft-search/pnp-modern-search#2906] [Q&A]
Description: We created a new Org member-no lIcense-ie since we will include External Users also for access to our app. Admin can run the page with PnP Search/Search Results and embedded PowerApps ChatGPT-3 connector but the app breaks when Org member runs IDENTICAL APP/SAME LINK! Clearly-it's not the app or PnP but Permission Level that is the issue. Need help to resolve-see attachments-shows screen shots for both Admin (rbalani@fdasmarts.onmicrosoft.com) and also Org Member (aditya2@fdasmarts.onmicrosoft.com) - Admin works but breaks when Org member logs-in.
See attachments:
1st one shows Admin working fine:
![Admin-Ram-SmartSearchPnP Works-Identical URL-Permission based issue](https://user-images.githubusercontent.com/37209670/231509186-3762bef8-73da-4091-8c4b-72e55edd542a.jpg)
2nd atatchment shows an Org Member (aditya2@fdasmarts.onmicrosoft.com) and the same PnP Search Web part breaking 
![Requiere JS-Error when Loading Pnp Search Web Part but Works for Admin](https://user-images.githubusercontent.com/37209670/231510507-32827ada-d1f1-4e37-b44b-51b50f3a2b30.jpg)

SIne the same Pnp Search Web part works when Admin runs it BUT the same breaks when Org member ( or also External Guest) this is clearly a PERMISSION based issues related to Pnp Modern Search Web part.
What are we doing wrong? What additional permission should be granted ?

**Comment by wobba** [comment]
See if any of the tips at https://laurakokkarinen.com/sharepoint-online-guest-user-troubles-and-how-to-get-past-them/ can help you out.

**Comment by RamBalani** [comment]
Thanks a bunch—will check out.

Buying you coffee mate.☕



Best Regards,
Ram Balani
CEO - eSTARHelper LLC

[cid:5e4f4cf2-b0bd-41ba-bb81-edbafd737df8]
________________________________

**Comment by RamBalani** [comment]
Mikael,

Thanks again for assisting with our Microsoft  'Prepared to be rescued' outreach.

Will investigate article you shared with us with our tech team —but it seems CDN access denied is a possible cause for our things breaking up all of a sudden when formerly the same PnP Modern Search app was working, even for External Guest! Strange indeed.

Let me clarify—recently our PowerApps Trial connector to OpenAI ChatGPT-3 (Powerapps embedded in the SmartSearch+ app-one and the same Communication page) expired—so we signed for PowerApps per App Premium subscription.

If I recal correctly—things seem to start falling apart then when in fact during the Premium Connector trial period-the PnP Modern Search web part we deployed with SmartSearch+  worked just fine! even for External Guests !

https://tinyurl.com/3jbjcd7v         The left-hand side mobile phone app screen image on this link is your awesome Pnp Modern Search Web part we customized then deployed with verticals context aware US FDA website content, i.e we've' collected all 2600 PDF Guidance PDFs and over 25K (SPO List items)  21 CFRs (Excels turned into SPO Lists) and rendered them searchable within PnP Search Web part as part of SmartSearch+!

I believe the fact that ChatGPT-3 embedded below the PnP Modern Search web part as shown in the attachment might be culprit !

The error message in the attachment points  to CDN references we get seems to me   highly suggestive of the source of misfires with us.

BTW
NIH (National Instiute of Health) and US FDA are both  interested with SmartSearch+ -i.e if deployed—could a great showcase for Microsoft SPO/PnP Search !

 And by that I mean - enhancing US FDA content search beyond what is achievable with US FDA's own website search and  even with pre-trained OpenAI ChatGPT-3 LLM model, i.e. a full-text search on actual FDA content as we have done with SmartSearch+  can never be outdone by an OpenAI (ChatGPT-xx) domain trained LLM model!
(But then again you already know this-what with your mastery of Microsoft Search !)

Best Regards,
Ram Balani
CEO - eSTARHelper LLC

[cid:eb7745ac-9ff2-41aa-a09f-27ad8dc1274c]

________________________________
